drivers/mdio/Kconfig.stm32_hal

```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2024 BayLibre, SAS
+# Copyright (c) 2010-2024 Analog Devices Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+config MDIO_ST_STM32_HAL
+	bool "STM32 MDIO controller driver"
+	default y
+	depends on SOC_SERIES_STM32H7X || SOC_SERIES_STM32H5X || ETH_STM32_HAL_API_V2
```

Comment:
is the dependency on the SOC kconfigs required if it already depends on the devicetree having the node enabled? presumably, a correct devictree would not have this node if the SOC does not contain the IP, no?

Reply:
soubds correct. fixed